Fix: Enable/Disable toggle does not disable all inputs (#284)

singhmahipal opened the pull request:
Related issue: #284

This PR fixes the issue where inputs and actions remain interactive
even when the extension is disabled.

## Summary by Sourcery

Ensure the popup UI fully respects the global enable/disable toggle and hardens popup script interactions against missing DOM elements.

Bug Fixes:
- Prevent platform dropdown, repository search/filter controls, token visibility toggle, checkboxes, radios, date range inputs, and scrum report content from remaining interactive when the extension is disabled.

Enhancements:
- Add a cached enable/disable state guard and ARIA/disabled styling so disabled controls are visually distinct, non-interactive, and more accessible.
- Make popup initialization and event wiring resilient to missing elements by adding null checks around DOM lookups.

singhmahipal commented:
@sourcery-ai review

singhmahipal commented:
@sourcery-ai review

singhmahipal commented:
Thanks for the review @vedansh-5 . I’ve addressed Sourcery’s concerns and made the suggested changes.

vedansh-5 commented on src/popup.html:
```diff
@@ -37,6 +37,80 @@
+
+		#githubToken[disabled],
+		#toggleTokenVisibility[disabled] {
+			cursor: not-allowed !important;
```
Please add all this css in index.css file

vedansh-5 reviewed (changes requested):
Turn DEBUG off in scrumHelper.js. Ott lgtm Thanks!

singhmahipal commented:
Hi @vedansh-5 ,

I’ve resolved the requested changes:
* All CSS for disabled/aria-disabled states has been moved from `popup.html` to `index.css`.
* `DEBUG` in `scrumHelper.js` has been turned off.

Thanks for the guidance and review — really appreciate it! 🙌

Everything is now updated and ready for your review.

vedansh-5 reviewed (commented):
@singhmahipal  A small change, the color of the buttons in first picture should be the same as that of the button in second picture.
<img width="433" height="374" alt="image" src="https://github.com/user-attachments/assets/cf27f3b0-4dbc-4696-991a-034d20ced199" />
<img width="421" height="605" alt="image" src="https://github.com/user-attachments/assets/9acd7cd0-b955-428a-bd16-93a0d8679d35" />
Also please make sure the color of the client dropdown matches the background in dark mode. 
<img width="410" height="391" alt="image" src="https://github.com/user-attachments/assets/960811f7-caf6-484d-9acb-9ae35d682c6e" />

OTT good to merge

singhmahipal commented:
Addressed the requested UI tweaks:


* Updated the buttons in the first view to match the color used in the second view.

<img width="440" height="358" alt="Screenshot 2026-01-07 170725" src="https://github.com/user-attachments/assets/ac3be7b6-9518-417e-838b-7bc87b678dd3" />



* Ensured the client dropdown background correctly matches the dark mode theme.

<img width="461" height="349" alt="Screenshot 2026-01-07 170705" src="https://github.com/user-attachments/assets/5fdf6554-4b5f-4791-ba45-2ce03a1a9289" />


and now it should be ready to merge.

hpdang commented:
@singhmahipal cld you please resolve the conflicts. Thanks

singhmahipal commented:
Hi @hpdang / @vedansh-5 👋  
All conflicts are resolved and the branch is up to date with master.  
There is 1 workflow awaiting maintainer approval.  
Could you please approve the workflow so this PR can be merged?  
Thanks!

vedansh-5 commented:
Thanks @singhmahipal for your contribution. Over the past weeks we have moved over to a new direction where we are planning to remove the on/off toggle for the extension. Therefore this PR no longer holds good. I am closing this. Please take up another issue from [issues tab](https://github.com/fossasia/scrum_helper/issues).
